src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -88,22 +101,77 @@ def __call__(self,
 
         serialized_request = _common.serialize(request,
                                                self._request_serializer)
-        timeout = self._timeout_to_deadline(timeout)
         aio_cancel_status = cygrpc.AioCancelStatus()
-        aio_call = asyncio.ensure_future(
-            self._channel.unary_unary(self._method, serialized_request, timeout,
-                                      aio_cancel_status),
-            loop=self._loop)
+
+        if self._interceptors:
+            client_call_details = _ClientCallDetails(
+                self._method, timeout, metadata, credentials, wait_for_ready,
+                compression)
+            aio_call = asyncio.ensure_future(
+                self._wrap_call_in_interceptors(
+                    client_call_details, serialized_request, aio_cancel_status),
+                loop=self._loop)
```

Comment:
Optional: Should we investigate if the `contextvars` are still plumbed through, if we create multiple layers of `Task` or `Future`?

Reply:
Yes, new tasks automatically inherit all context from the parent tasks [1].

For futures would depend on the task context where a future is being accessed, but for the relevant part the following code:

```python
    await future
```

The task awaiting for this future would be bound to the task context, nothing else.

[1] https://github.com/python/cpython/blob/edad4d89e357c92f70c0324b937845d652b20afd/Lib/asyncio/tasks.py#L164

Reply:
This is great!